src/Mapper.php

```diff
@@ -1,11 +1,156 @@
 <?php
 
-namespace Tnapf\JsonMapper\Mapper;
+namespace Tnapf\JsonMapper;
+
+use ReflectionAttribute;
+use ReflectionClass;
+use ReflectionProperty;
+use Tnapf\JsonMapper\Attributes\AnyArray;
+use Tnapf\JsonMapper\Attributes\AnyType;
+use Tnapf\JsonMapper\Attributes\CaseConversionInterface;
+use Tnapf\JsonMapper\Attributes\BaseType;
+use Tnapf\JsonMapper\Attributes\BoolType;
+use Tnapf\JsonMapper\Attributes\FloatType;
+use Tnapf\JsonMapper\Attributes\IntType;
+use Tnapf\JsonMapper\Attributes\ObjectArrayType;
+use Tnapf\JsonMapper\Attributes\ObjectType;
+use Tnapf\JsonMapper\Attributes\StringType;
 
 class Mapper implements MapperInterface
 {
+    protected ?CaseConversionInterface $caseConversion = null;
+    protected object $instance;
+    protected ReflectionClass $reflection;
+
+    /**
+     * @var BaseType[]
+     */
+    protected array $attributes;
+
+    protected string $class;
+    protected array $data;
+
     public function map(string $class, array $data): object
     {
-        $instance = new $class();
+        $instance = (new self());
+        $instance->class = $class;
+        $instance->data = $data;
+        $instance->reflection = new ReflectionClass($class);
+
+        if ($attributes = $instance->reflection->getAttributes(CaseConversionInterface::class, ReflectionAttribute::IS_INSTANCEOF)) {
+            if (count($attributes) > 1) {
+                throw new MapperException("{$class} has more than one case conversion attribute");
+            }
+
+            $instance->caseConversion = $attributes[0]->newInstance();
+        }
+
+        $instance->instance = new $class();
```

Comment:
You most probably will encounter a situation where the mapped object has a constructor with required arguments, so this will fail. I recommend using `ReflectionClass::newInstanceWithoutConstructor()` to create an instance of mapped object instead of `new $class();`.

Reply:
But isn't that as bad as using eval?

Reply:
Something that requires mapping doesn't actually carry any serious logic in the constructor. Even if it does, it's not the problem of the mapper, it's the problem of the mapped class.

And on `eval` - reflection itself is bad and should be avoided. This lib is using meta information about mapped object, so, naturally, it has all of the reflection potential at its disposal.

Reply:
Fair point, will add